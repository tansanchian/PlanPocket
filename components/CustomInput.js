import {
  Text,
  View,
  TextInput,
  StyleSheet,
  Button,
  Platform,
  TouchableOpacity,
} from "react-native";
import React, { useState } from "react";
import { Controller } from "react-hook-form";
import DateTimePicker from "@react-native-community/datetimepicker";

const CustomInput = ({
  control,
  rules = {},
  name,
  placeholder,
  secureTextEntry,
  design = "",
  keyboard,
}) => {
  const [date, setDate] = useState(new Date());
  const [showDatePicker, setShowDatePicker] = useState(false);

  const onChanges = (event, selectedDate) => {
    const currentDate = selectedDate || date;
    setShowDatePicker(Platform.OS === "ios");
    setDate(currentDate);
  };

  const showDatepicker = () => {
    setShowDatePicker(true);
  };
  if (keyboard == "time") {
    return (
      <TouchableOpacity
        style={[styles[`container${design}`]]}
        onPress={showDatepicker}
      >
        <View pointerEvents="none">
          <TextInput
            value={date.toLocaleTimeString([], {
              hour: "2-digit",
              minute: "2-digit",
            })}
            onChangeText={onChanges}
          />
          {showDatePicker && (
            <DateTimePicker value={date} mode="time" onChange={onChanges} />
          )}
        </View>
      </TouchableOpacity>
    );
  } else if (keyboard == "date") {
    return (
      <TouchableOpacity
        style={[styles[`container${design}`]]}
        onPress={showDatepicker}
      >
        <View pointerEvents="none">
          <TextInput
            value={date.toLocaleDateString()}
            onChangeText={onChanges}
          />
          {showDatePicker && (
            <DateTimePicker value={date} mode="date" onChange={onChanges} />
          )}
        </View>
      </TouchableOpacity>
    );
  } else {
    return (
      <Controller
        control={control}
        name={name}
        rules={rules}
        render={({
          field: { value, onChange, onBlur },
          fieldState: { error },
        }) => (
          <>
            <View
              style={[
                styles[`container${design}`],
                { borderColor: error ? "red" : "#e8e8e8" },
              ]}
            >
              <TextInput
                style={styles.input}
                value={value}
                onBlur={onBlur}
                onChangeText={onChange}
                placeholder={placeholder}
                secureTextEntry={secureTextEntry}
              />
            </View>
            {error && (
              <Text style={styles.errorText}>{error.message || "Error"}</Text>
            )}
          </>
        )}
      />
    );
  }
};

export default CustomInput;

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    width: "100%",
    borderColor: "#e8e8e8",
    borderWidth: 1,
    borderRadius: 5,
    paddingVertical: 5,
    paddingHorizontal: 10,
    marginVertical: 5,
  },
  containerHALF: {
    backgroundColor: "white",
    width: "50%",
    borderColor: "#e8e8e8",
    borderWidth: 1,
    borderRadius: 5,
    paddingVertical: 5,
    paddingHorizontal: 10,
    marginVertical: 5,
  },
  input: {},
  errorText: {
    color: "red",
    alignSelf: "stretch",
    textAlignVertical: "bottom",
  },
});
