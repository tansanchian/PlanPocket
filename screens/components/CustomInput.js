import { Text, View, TextInput, StyleSheet } from "react-native";
import React, { useState } from "react";
import { Controller } from "react-hook-form";

const CustomInput = ({
  control,
  rules = {},
  name,
  placeholder,
  secureTextEntry,
  design = "",
  keyboard,
  editable = true,
  values = "",
}) => {
  return (
    <Controller
      control={control}
      name={name}
      rules={rules}
      render={({
        field: { value = values, onChange, onBlur },
        fieldState: { error },
      }) => (
        <>
          <View
            style={[
              styles[`container${design}`],
              { borderColor: error ? "red" : "#e8e8e8" },
            ]}
          >
            <TextInput
              style={styles.input}
              value={value}
              onBlur={onBlur}
              onChangeText={onChange}
              placeholder={placeholder}
              secureTextEntry={secureTextEntry}
              editable={editable}
              keyboardType={keyboard}
            />
          </View>
          {error && (
            <Text style={styles.errorText}>{error.message || "Error"}</Text>
          )}
        </>
      )}
    />
  );
};

export default CustomInput;

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    width: "100%",

    borderColor: "#e8e8e8",
    borderWidth: 1,
    borderRadius: 5,
    paddingVertical: 5,
    paddingHorizontal: 10,
    marginVertical: 5,
    borderRadius: 25,
  },
  containerHALF: {
    backgroundColor: "white",
    width: "50%",
    borderColor: "#e8e8e8",
    borderWidth: 1,
    borderRadius: 5,
    paddingVertical: 5,
    paddingHorizontal: 10,
    marginVertical: 5,
    borderRadius: 25,
  },
  input: {},
  errorText: {
    color: "red",
    alignSelf: "stretch",
    textAlignVertical: "bottom",
    borderRadius: 25,
  },
});