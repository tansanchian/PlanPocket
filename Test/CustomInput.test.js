import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { useForm } from "react-hook-form";
import CustomInput from "../components/CustomInput";

const MockForm = ({
  rules,
  name,
  placeholder,
  secureTextEntry,
  design,
  keyboard,
  editable,
  values,
}) => {
  const { control } = useForm();
  return (
    <CustomInput
      control={control}
      rules={rules}
      name={name}
      placeholder={placeholder}
      secureTextEntry={secureTextEntry}
      design={design}
      keyboard={keyboard}
      editable={editable}
      values={values}
    />
  );
};

describe("CustomInput", () => {
  it("renders correctly and matches snapshot", async () => {
    const { toJSON } = render(
      <MockForm name="test" placeholder="Enter text" />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  // it("displays an error message when validation fails", () => {
  //   const { getByPlaceholderText, findByText } = render(
  //     <MockForm
  //       name="test"
  //       placeholder="Enter text"
  //       rules={{ required: "This field is required" }}
  //     />
  //   );

  //   const input = getByPlaceholderText("Enter text");
  //   fireEvent.changeText(input, "");

  //   const errorMessage = findByText("This field is required");
  //   expect(errorMessage).toBeTruthy();
  // });

  it("handles user input correctly", () => {
    const { getByPlaceholderText } = render(
      <MockForm name="test" placeholder="Enter text" />
    );

    const input = getByPlaceholderText("Enter text");
    fireEvent.changeText(input, "Test input");

    expect(input.props.value).toBe("Test input");
  });
});
