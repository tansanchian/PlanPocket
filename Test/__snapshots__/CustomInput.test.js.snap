// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders correctly and matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderColor": "#e8e8e8",
        "borderRadius": 25,
        "borderWidth": 1,
        "elevation": 5,
        "marginVertical": 5,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 8,
        "width": "100%",
      },
      {
        "borderColor": "#e8e8e8",
      },
    ]
  }
>
  <TextInput
    editable={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Enter text"
    style={{}}
    value=""
  />
</View>
`;
